server/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java

```diff
@@ -128,4 +131,43 @@ public void aWaitingTaskShouldBeCancelled(CountDownLatch countDownLatch) {
     @Disabled("Cancelling is not supported yet")
     public void getStatusShouldBeCancelledWhenCancelled(CountDownLatch countDownLatch) {
     }
+
+    @Test
+    void givenTwoTaskManagersAndTwoTaskOnlyOneTaskShouldRunAtTheSameTime() throws InterruptedException {
+        CountDownLatch waitingForFirstTaskLatch = new CountDownLatch(1);
+
+        TaskManager taskManager1 = taskManager();
+        TaskManager taskManager2 = taskManager();
+
+        taskManager1.submit(new MemoryReferenceTask(() -> {
+            waitingForFirstTaskLatch.await();
+            return Task.Result.COMPLETED;
+        }));
+        TaskId waitingTaskId = taskManager1.submit(new CompletedTask());
+        Thread.sleep(100);
+
+        assertThat(taskManager2.getExecutionDetails(waitingTaskId).getStatus()).isEqualTo(TaskManager.Status.WAITING);
+        waitingForFirstTaskLatch.countDown();
+
+        Awaitility.await()
+            .atMost(Duration.ONE_SECOND)
+            .pollInterval(100L, TimeUnit.MILLISECONDS)
+            .until(() -> taskManager1.await(waitingTaskId).getStatus() == TaskManager.Status.COMPLETED);
+    }
+
+    @Test
+        // FIXME it's currently dependent of the implementation of the sequential TaskManager with the exclusive RabbitMQ consumer
+        // once we store the node where the event have been created/started/completed we should rewrite it with this information.
+    void givenTwoTaskManagerATaskSubmittedOnOneCouldBeRunOnTheOther() throws InterruptedException {
+        TaskManager taskManager1 = taskManager();
+        Thread.sleep(100); // FIXME used to ensure that taskManager1 is the worker consuming from rabbit
```

Comment:
you actually register to rabbit events for that, I guess.
However, not sure it worth the investment

Reply:
I propose a ticket for that, I think it will be nice to have it

Reply:
do you have a link to this ticket?

Reply:
https://ci.linagora.com/linagora/lgs/openpaas/james/issues/1353